src/pudl/output/eia860.py

```diff
@@ -136,11 +138,168 @@ def plants_eia860(pudl_engine, start_date=None, end_date=None):
+    )
+    return plants
+
+
+def fill_in_missing_ba_codes(plants: pd.DataFrame) -> pd.DataFrame:
+    """Fill in missing ``balancing_authority_code_eia`` using either bfill or most consistent.
+
+    Balancing authority codes did not begin being reported until 2013. This function
+    fills in the old years with BA codes using two main methods:
+
+    * Backfilling from the oldest reported BA code (via :meth:`pd.bfill`)
+    * Using the most consistent reported value
+
+    We add a column to represent each of these two methodologies via
+    :func:`make_backfilled_ba_code_column` and :func:`make_consistent_ba_code_column`.
+
+    We know that the BA codes do change over time and are incorrectly reported at times.
+    Because of these two facts, we cann't simple :meth:`pd.fillna` with either the
+    backfilled or most consistent value. This function employs some specific filling in
+    methodologies based on an investigation of the data. Here is a description of each
+    of various stages of filling in which involves some bespoke cleanup:
+
+    * if the backfilled code and the most consistent code are the same, use either!
+    * assign BA code ``PACW`` (``PacifiCorp - West``) when ``PACW`` is the most
+      consistent BA code and the ``state`` is within ``PACW``'s territory
+    * similarity, assign BA code ``PACE`` (``PacifiCorp - East``) when ``PACE`` is
+      the most consistent BA code and the ``state`` is within ``PACE``'s territory
+    * use the backfilled code for plants that have ``SWPP`` (``Southwest Power Pool``)
+      as their most consistent BA code because we know ``SWPP`` accumulated many smaller
+      balancing authorities
+    * use the backfilled code for plants that have a differnt backfilled and most
+      consistent code, but the most consistent code is has a consistency rate (how
+      frequent the most consistent BA code is reported / the total BA codes reported)
+      is less than 80%. This is meant to find plants where there are multiple years of
+      an older BA code before it switches - assuming a BA code being reported for
+      multiple years is not a reporting error. TODO: make this one more explicit.
+
+    Args:
+        plants: table of annual plant attributes, including ``balancing_authority_code_eia``
+    """
+
+    def log_current_ba_code_nulls(plants: pd.DataFrame, method_str: str) -> None:
+        """Internal function to log progress on fillin in BA codes.
+
+        Args:
+            plants: the current plants table to check
+            method_str: A description of the method employed. This will be inserted into
+                the log.
+        """
+        currently_null_len = len(plants[plants.balancing_authority_code_eia.isnull()])
+        logger.info(
+            f"Filled BA codes where {method_str}. "
+            f"Currently {currently_null_len/len(plants):.1%} of records ({currently_null_len}) with no BA codes"
+        )
```

Comment:
I **totally** get the desire to functionalize this, but the messages seem confusing to me. I think it would be clearer if you got rid of the prefix "Filled BA codes where..."  and remove "Currently" and change "with" to "have" in the suffix.

Also providing only 1 number is unclear to me -- is that the total number of records? Or is that the number of records which corresponds to the percentage? I'd either provide just the percentage, the percentage with the complete fraction (missing/total), or just the fraction (missing/total). E.g.

> Filled BA codes where no treatment had been applied. Currently 43.9% of records (81266) with no BA codes

"Before any filling treatment has been applied 43.9% of records have no BA code."

Reply:
Remove the "filled.." part make 100% sense to me - especially after i started using it b4 any filling happened. 

I can remove one of the numbers. I'd rather keep the % bc it feels way more useful. Although I'll have to up the sig figs bc the late-stage filling effects very few # of records. Also to be fair, I was copying your logs from the bfill process.
```
Started with 81266 missing BA Codes out of 185071 records (43.91%)
Ended with 14923 missing BA Codes out of 185071 records (8.06%)
```

Reply:
Both numbers is also great! It's just that without both the numerator and denominator it's not clear what absolute number corresponds to the percentage. The EIA 861 BA code logging message includes the both (though they aren't right next to each other). "81266 (out of) 185071 records"

Reply:
i think both numbers are duplicative with the %. i'd rather take the # value out than have both.